source/20_spec/config_spec.rst

```diff
@@ -0,0 +1,154 @@
+
+-  Ensure evaluator_env_num：n_evaluator_episode = 1:1 （expect smac environment）
+
+-  manager field shoudl generally not be included in the env field of main_config
+   (shared_memory defaults to True when the manager field is not included)：
+
+   -  smac environment is an exception,due to the state dimension problem,smac needs to set shared_memory=Fasle。
+
+   -  In environments other than the smac environment, if an error is reported due to the state dimension problem,you can include manager field and set  shared
+      memory=False。
+
+-  If you want to turn on/off shared memory, please control it in env.manager filed
+
+   .. code:: python
+
+      config = dict(
+          ...,
+          env=dict(
+              manager=dict(
+                  shared_memory=True,
+              ),
+          ),
+      )
+
+-  create config
+
+   -  in env field,only ``type`` 和 ``import_names``\ two fields,
+      Such as：
+
+   .. code:: python
+
+      env=dict(
+          type='atari',
+          import_names=['dizoo.atari.envs.atari_env'],
+      ),
+
+   -  nnormally field \ ``replay_buffer``\ is unnecessary。If you want to use the buffer stored as deque，please specify the type of replay_buffer：
+
+      .. code::
+
+         replay_buffer=dict(type='deque'),
+
+-  serial_pipeline
+
+   -  Please apple secondary references to avoid circular
+      import：use \ ``from ding.entry import serial_pipeline``\ instead of \ ``from ding.entry.serial_entry import serial_pipeline``
+
+   -  Use\ ``[main_config, create_config]``
+      to unify the style,If an algorithm needs to call other config,this convention can be waived。Such as imitation
+      learning algorithm needs to introduce expert config, see the example of sqil for details。
+
+   -  Every config must have a starting command,and it's format should as below
+
+      .. code:: python
+
+         if ___name___ == "___main___":
+             # or you can enter `ding -m serial -c cartpole_dqn_config.py -s 0`
+             from ding.entry import serial_pipeline
+             serial_pipeline([main_config, create_config], seed=0)
+
+      -  Remember \ ``from ding.entry import serial_pipeline``\ this line should not as the head of the file,
+         please note it at \ ``if ___name___ == "___main___":``\ below.
+
+   -  If the algorithm use different serial_pipeline_X,
+      you need to add corresponding starting command ``serial_X``\ in \ https://github.com/opendilab/DI-engine/blob/5d2beed4a8a07fb70599d910c6d53cf5157b133b/ding/entry/cli.py#L189\ .
+
+-  seed is set in the entry function, do not include seed in config.
+
+-  If the hyperparameters in the algorithm have a certain reasonable range, please write a comment on the corresponding hyperparameters of the algorithm config. For instance the alpha value of sqil：
+
+   .. code:: python
+
+      alpha=0.1,  # alpha: 0.08-0.12
+
+-  Please make sure all parameters in config are valid ,unused keys should be deleted.
```

Comment:
Please make sure all parameters in config are valid, unused redundant  parameters should be deleted.